### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by AbdelazizSoliman**
In this pull request I updateded  model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png). and created a custom methods  for models.

### CODE REVIEW

File: app/models/user.rb
Comment (fobadara) [inline app/models/user.rb:1]: [REQUIRED] KINDLY INCLUDE VALIDATIONS FOR THIS FILE AS INSTRUCTED. I CANT SEEM TO FIND ANY VALIDATIONS. IT SHOULD  LOOK SOMETHING LIKE THIS: `validates :name, presence: true`.

File: app/models/post.rb
Comment (fobadara) [inline app/models/post.rb:1]: [REQUIRED] KINDLY INCLUDE VALIDATIONS FOR THIS FILE AS INSTRUCTED. I CANT SEEM TO FIND ANY VALIDATIONS. IT SHOULD LOOK SOMETHING LIKE THIS: validates :name, presence: true.

File: app/models/like.rb
Comment (fobadara) [inline app/models/like.rb:1]: [REQUIRED] KINDLY INCLUDE VALIDATIONS FOR THIS FILE AS INSTRUCTED. I CANT SEEM TO FIND ANY VALIDATIONS. IT SHOULD LOOK SOMETHING LIKE THIS: validates :name, presence: true.

File: app/models/comment.rb
Comment (fobadara) [inline app/models/comment.rb:1]: [REQUIRED] KINDLY INCLUDE VALIDATIONS FOR THIS FILE AS INSTRUCTED. I CANT SEEM TO FIND ANY VALIDATIONS. IT SHOULD LOOK SOMETHING LIKE THIS: validates :name, presence: true.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @AbdelazizSoliman, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary

## Required Changes ♻️
_Check the comments under the review._
[REQUIRED] KINDLY INCLUDE SPECS AS IT IS A REQUIREMENT FOR THIS MILESTONE.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AbdelazizSoliman** [comment]
Hello @fobadara , this pull request for processing data in models, and I submitted for validation by mistake, thank you.

**Review by rotimiazeez — APPROVED** [review]
Hi @AbdelazizSoliman 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Good job setting up the association between models correctly 🎉
* `User` model has a method that returns 3 most recent posts 👍🏽
* `Post` model has a method that returns the 5 most recent comments for a given post. 👌🏽
* Good job using `Rails Active Callbacks` ✌🏽
* Good and Descriptive PR 😉
- Good commit message(s) 👍🏽
- Descriptive README 📖

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AbdelazizSoliman** [comment]
Thank you @rotimiazeez.

**Pull request merged**